### PR TITLE
fix error handling for upload survey

### PR DESCRIPTION
**Pull request by rhiGriff**

### CODE REVIEW

File: frontstage/views/surveys.py [hunk 1]
```diff
@@ -241,6 +241,9 @@ def upload_survey(session):
         return render_template('surveys/surveys-upload-success.html', _theme='default', upload_filename=upload_filename)
     else:
```
Comment (andrewmil) [inline frontstage/views/surveys.py]: We probably want to log out something from the failed request. Maybe the status code at least?

File: frontstage/views/surveys.py [hunk 2]
```diff
@@ -241,6 +241,9 @@ def upload_survey(session):
...
         logger.debug('Upload failed')
```
Comment (andrewmil) [inline frontstage/views/surveys.py]: This should be an error log

File: frontstage/views/surveys.py [hunk 3]
```diff
@@ -241,6 +241,9 @@ def upload_survey(session):
...
-        error_info = json.loads(result.text)
+        if result.text is None:
+            logger.error('Upload failed due to error')
```
Comment (andrewmil) [inline frontstage/views/surveys.py]: This log is just saying the same as the previous one

File: frontstage/views/surveys.py [hunk 4]
```diff
@@ -241,6 +241,9 @@ def upload_survey(session):
...
+        else:
+            error_info = json.loads(result.text)
         return render_template('surveys/surveys-upload-failure.html',  _theme='default', error_info=error_info,
```
Comment (andrewmil) [inline frontstage/views/surveys.py]: will this render_template work if no error_info has been defined?, i.e. if result.text=None

File: frontstage/views/surveys.py [hunk 5]
```diff
@@ -240,10 +240,13 @@ def upload_survey(session):
         logger.debug('Upload successful')
         return render_template('surveys/surveys-upload-success.html', _theme='default', upload_filename=upload_filename)
     else:
-        logger.debug('Upload failed')
-        if result.text is None:
+        logger.error('Upload failed. url: {}, status code: {}'.format(url, result.status_code))
```
Comment (andrewmil) [inline frontstage/views/surveys.py]: We're probably going to change the formatting of logs in an upcoming card maybe but the content here all looks good

**Comment by rhiGriff** [comment]
Waiting for designs to come back before progressing with this PR

File: frontstage/views/surveys.py [hunk 6]
```diff
@@ -247,7 +247,7 @@ def upload_survey(session):
         logger.debug('"event" : "Upload successful"')
         return render_template('surveys/surveys-upload-success.html', _theme='default', upload_filename=upload_filename)
     else:
-        logger.debug('"event" : "Upload failed"')
-        error_info = json.loads(result.text)
-        return render_template('surveys/surveys-upload-failure.html',  _theme='default', error_info=error_info,
-                               case_id=case_id)
+        logger.error('Upload failed. url: {}, status code: {}'.format(url, result.status_code))
+        error_info = {'status code': result.status_code,
+                      'text': result.text}
```
Comment (rjweeks) [inline frontstage/views/surveys.py]: Put this on one line please.

**Comment by rjweeks** [comment]
Can you update to master.

File: frontstage/views/surveys.py [hunk 7]
```diff
@@ -247,7 +247,6 @@ def upload_survey(session):
         logger.debug('"event" : "Upload successful"')
         return render_template('surveys/surveys-upload-success.html', _theme='default', upload_filename=upload_filename)
     else:
-        logger.debug('"event" : "Upload failed"')
-        error_info = json.loads(result.text)
-        return render_template('surveys/surveys-upload-failure.html',  _theme='default', error_info=error_info,
-                               case_id=case_id)
+        logger.error('Upload failed. url: {}, status code: {}'.format(url, result.status_code))
+        error_info = {'status code': result.status_code, 'text': result.text}
+        return render_template('surveys/surveys-upload-failure.html', _theme='default', error_info=error_info, case_id=case_id)
```
Comment (andrewmil) [inline frontstage/views/surveys.py]: According to coverage this code isn't tested but that could be done in a separate PR probably as it wasn't done before/is subject to change

**Pull request merged**